Perl/syntax_test_perl.pl

```diff
@@ -787,6 +787,39 @@ =head1 B<--param>
 #         ^^^^^ string.quoted.double.perl
 #             ^ punctuation.definition.string.end.perl
 #              ^ punctuation.terminator.statement.perl
+  $foo{bar}
+# ^^^^ variable.other.readwrite.global.perl
+# ^ punctuation.definition.variable.perl
+#     ^ punctuation.section.item-access.begin.perl
+#     ^^^^^ meta.item-access.perl
+#      ^^^ string.unquoted.perl
+#         ^ punctuation.section.item-access.end.perl
+  $foo{bar()}
+# ^^^^ variable.other.readwrite.global.perl
+# ^ punctuation.definition.variable.perl
+#     ^ punctuation.section.item-access.begin.perl
+#     ^^^^^ meta.item-access.perl
+#      ^^^ variable.function.perl
+#           ^ punctuation.section.item-access.end.perl
+  $foo{10 + $bar}
+# ^^^^ variable.other.readwrite.global.perl
+# ^ punctuation.definition.variable.perl
+#     ^ punctuation.section.item-access.begin.perl
+#     ^^^^^^^^^^^ meta.item-access.perl
+#      ^^ constant.numeric.integer.decimal.perl
+#         ^ keyword.operator.arithmetic.perl
+#           ^^^^ variable.other.readwrite.global.perl
+#               ^ punctuation.section.item-access.end.perl
+  %{$foo{bar}{baz}} = 'excl';
```

Comment:
Wouldn't you want a test on `bar` or `baz` here as well? Otherwise LGTM.

Reply:
Reasonable. Missed that in the quoted counter part as well.